BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -337,10 +337,19 @@ public void SaveFiles(List<IFormFile> files)
                 {                    
                     using (Stream uploadedFileStream = file.OpenReadStream())
                     {
-                        byte[] buffer = new byte[uploadedFileStream.Length];
-                        uploadedFileStream.Read(buffer, 0, buffer.Length);
                         string name = CreateFileName(file.FileName);
-                        _fileSystem.WriteFile($"{UploadsFolder}\\{name}", buffer);
+                        byte[] buffer = new byte[1024];
+                        int index = 0;
+                        while(index < buffer.Length)

```

Comment:
Won't this break the loop as soon as you've read the first 1024 bytes (i.e. index += 1024 down below?